src/Dotnet.Script/Program.cs

```diff
@@ -133,17 +133,20 @@ private static int Wain(string[] args)
                 c.Description = "Creates an executable from a script";
                 var fileNameArgument = c.Argument("filename", "The script file name");
                 var publishDirectoryOption = c.Option("-o |--output", "Directory where the published executable should be placed.  Defaults to a 'publish' folder in the current directory.", CommandOptionType.SingleValue);
+                var dllOption = c.Option("--dll", "Publish to a .dll instead of a .exe", CommandOptionType.NoValue);
+                var commandConfig = c.Option("-c | --configuration <configuration>", "Configuration to use for running the script [Release/Debug] Default is \"Debug\"", CommandOptionType.SingleValue);
```

Comment:
why do we need an extra option instead of the already existing top-level ` var configuration = app.Option("-c | --configuration <configuration>"`?

Reply:
I'm glad you mentioned this. I tested trying to access "app" level "options" when executing a sub "command", and those options were not coming through and required me to create new variables in order to access them at the "command" level.

Reply:
I am barely able to handle this argument parsing library 😀 so I now suspect this is happening because the `publish` is a command so it creates a new context and all the arguments are parsed within this context, not in the global one?

Reply:
Yeah that has been the assumption I went with as well.